prompb/remote.proto

```diff
@@ -23,10 +23,32 @@ message WriteRequest {
   repeated prometheus.TimeSeries timeseries = 1 [(gogoproto.nullable) = false];
 }
 
+// ReadRequest represents remote read request.
 message ReadRequest {
   repeated Query queries = 1;
+
+  enum ResponseType {
+    // Server will return ReadResponse with raw samples.
+    //
+    // Response headers:
+    // Content-Type: "application/x-protobuf"
+    // Content-Encoding: "snappy"
+    SAMPLES = 0;
+
+    // Server will stream varint delimited ChunkedReadResponse message that contains XOR encoded chunks for single series.
+    //
+    // Response headers:
+    // Content-Type: "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse"
+    // Content-Encoding: ""
+    STREAMED_XOR_CHUNKS = 1;
+  }
+  // response_type allows negotiating content type of response.
+  // This does not guarantee the returned response. For servers that does not support this field, SAMPLES response type
+  // is used.
+  ResponseType response_type = 2;
```

Comment:
Shouldn't this be a list, in order of preference?

Reply:
Very good question but will move this comment to https://github.com/prometheus/prometheus/pull/5701